### ISSUE TITLE
mlx_lm.lora training don't report anything

### ISSUE THREAD

**Issue by arthasyou** [ml-explore/mlx-examples#686]
```
{
    "adapter_path": "/Users/you/models/lora",
    "batch_size": 4,
    "config": null,
    "data": "/Users/you/data",
    "grad_checkpoint": false,
    "iters": 100,
    "learning_rate": 1e-05,
    "lora_layers": 16,
    "lora_parameters": {
        "rank": 8,
        "alpha": 16,
        "dropout": 0.0,
        "scale": 10.0
    },
    "lr_schedule": null,
    "max_seq_length": 2048,
    "model": "/Users/you/models/TinyLlama-1.1B-Chat-v1.0",
    "resume_adapter_file": null,
    "save_every": 100,
    "seed": 0,
    "steps_per_eval": 200,
    "steps_per_report": 10,
    "test": false,
    "test_batches": 500,
    "train": true,
    "val_batches": 25
}
```

The above is the configuration file. When running lora train, no message is output from the terminal after training for an hour. How can I make the terminal display relevant information? I want to know what the total training time is.

**Comment by awni** [comment]
Usually you should see something after a few seconds. The fact that you do not suggests that it's using way too much memory for your machine. There are a [few suggestions here](https://github.com/ml-explore/mlx-examples/blob/main/llms/mlx_lm/LORA.md#Memory-Issues) on reducing memory use.

Basically:
- Reduce the batch size
- Reduce the number of LoRA layers 
- Use gradient checkpointing
- Reduce the `max_seq_length`

**Comment by arthasyou** [comment]
Loading pretrained model
Trainable parameters: 0.074% (0.819M/1100.048M)
Loading datasets
Training
Starting training..., iters: 100

The above is all the information displayed by the terminal
My mac stuio has the highest configuration and has 192G memory.

**Comment by awni** [comment]
How odd.

**Comment by awni** [comment]
Can you share the version of MLX and MLX LM that you are using?

**Comment by arthasyou** [comment]
Loading pretrained model
Trainable parameters: 0.074% (0.819M/1100.048M)
Loading datasets
Training
Starting training..., iters: 100

**Comment by arthasyou** [comment]
mlx                0.10.0
mlx-lm             0.9.0

**Comment by arthasyou** [comment]
and my data just 1 message
{"prompt": "What is the capital of France?", "completion": "Paris."}
only for testing mlx lora trainning

**Comment by awni** [comment]
Problem is the `batch_size=4` and the dataset only has one example, so it's getting stuck in an infinite loop in the data construction. You can use a larger dataset or a batch size of 1 to get it to work immediately. 

It's a bit of an edge case but I'll try to push a fix for this so it's more clear what is going on.

**Comment by arthasyou** [comment]
thx so much it's working
